model/TestCenterService.php

```diff
@@ -25,14 +25,24 @@
 use core_kernel_classes_Resource;
 use oat\oatbox\user\User;
 use oat\tao\model\TaoOntology;
-use tao_models_classes_ClassService;
+use oat\tao\model\ClassServiceTrait;
+use oat\tao\model\GenerisServiceTrait;
+use oat\oatbox\service\ConfigurableService;
+use oat\oatbox\service\ServiceManager;
 
 /**
  * TestCenter Service for proctoring
  *
  */
-class TestCenterService extends tao_models_classes_ClassService
+class TestCenterService extends ConfigurableService
 {
+    use ClassServiceTrait {
+        deleteResource as protected traitDeleteResource;
```

Comment:
The delete resource method just call a `delete()` method on the given `$resource`. Maybe it could decrease the complexity just to call that method instead of loading the Trait class. What do you think?

Reply:
We need that trait for all the rest methods as well (such as deleteClass, deleteClassProperty, etc.).